pylint/utils/pragma_parser.py

```diff
@@ -9,12 +9,9 @@
 # so that an option can be continued with the reasons
 # why it is active or disabled.
 OPTION_RGX = r"""
-    \s*                # Any number of whitespace
-    \#?                # One or zero hash
-    .*                 # Anything (as much as possible)
-    (\s*               # Beginning of first matched group and any number of whitespaces
+    (                  # Beginning of first matched group and any number of whitespaces
     \#                 # Beginning of comment
-    .*?                # Anything (as little as possible)
+    \s*?               # Any whitespaces (as little as possible)
```

Comment:
This is a breaking change, because every changes is a breaking change (*heavy sigh*), but it's probably worth it. 

We could do:
```suggestion
    .*?               # Anything (as little as possible)
```
Until we release 3.0, but then we need a way to make the deprecation known. So should we warn the user if we match a pylint comment with ``.*?`` but not with ``\s*?`` ? it would actually decrease performance in 2.X, possibly by a lot [citation required].

Another solution is to consider that it was an unintended feature and just consider this a performance fix. How many users are going to be pissed off by this decision... ? I wish I knew.

Reply:
I don't think there is a real performance change if we make `.*?` a capturing group and check if it contains anything other than spaces whenever we use this regex pattern. That should be relatively straightforward.

Reply:
Wouldn't it solve the breaking change issue ?

Reply:
I don't think so. The breaking change would be to remove `.*` completely and replace it with nothing or with `\s*`. To deprecate this we can use a capturing group and emit a warning when we match anything other than spaces.

Reply:
But why would we be doing the breaking change if there is no performance issues if we keep the current behavior ?